### PR TITLE
Minor p2p updates

### PR DESCRIPTION
**Pull request by pavel-kokolemin**
Just some minor improvements in p2p types (removing some duplicate and unused types).

### CODE REVIEW

File: dns_server/Cargo.toml [hunk 1]
```diff
@@ -30,5 +30,8 @@ clap = { version = "4", features = ["derive"] }
 directories = "4.0"
 
 [dev-dependencies]
-rstest = "0.16"
+# Local dependencies
```
Comment (TheQuantumPhysicist) [inline dns_server/Cargo.toml:33]: You really don't need to write these comments... stating the obvious isn't usually a good practice in comments.

File: p2p/src/interface/types.rs [hunk 2]
```diff
@@ -13,12 +13,14 @@
...
+
+use crate::types::peer_id::PeerId;
+
+#[derive(Debug, Serialize)]
 pub struct ConnectedPeer {
-    // TODO: Replace String with actual type, once libp2p removed
-    pub peer_id: String,
+    pub peer_id: PeerId,
 
-    // TODO: Replace String with actual type, once libp2p removed
```
Comment (stanislav-tkach) [inline p2p/src/interface/types.rs:21]: Are there chances that we can replace this string with the actual type too?

**Review by stanislav-tkach — APPROVED** [review]

File: dns_server/Cargo.toml (hunk 1 above)
Comment (pavel-kokolemin) [reply dns_server/Cargo.toml]: OK, sure, removed.

File: p2p/src/interface/types.rs (hunk 2 above)
Comment (pavel-kokolemin) [reply p2p/src/interface/types.rs:21]: The problem is that we don't know the actual type.
For example here:
```
impl<T: ToString> From<&PeerContext<T>> for ConnectedPeer {
    fn from(context: &PeerContext<T>) -> Self {
        ConnectedPeer {
            peer_id: context.info.peer_id,
            address: context.address.to_string(),
            inbound: context.role == Role::Inbound,
            ban_score: context.score,
        }
    }
}
```

I suppose we could try to replace it with the `PeerAddress` type, which can be constructed from a generic network address (`PeerContext` would need to implement `serde::Serialize` then).

**Review by TheQuantumPhysicist — APPROVED** [review]

**Pull request merged**